### PR TITLE
Fixes climatology variable detection and adds tests

### PR DESCRIPTION
**Pull request by lukecampbell**
- Detecting a climatology variable is now exactly as it is described in
  §7.4 of CF.
- There can only be one variable describing the climatology bounds in a
  CF compliant dataset.
- It's simply whatever the time variable points to with the
  `climatology` attribute.
- It is NOT a geophysical variable

### CODE REVIEW

**Comment by lukecampbell** [comment]
Fixes https://github.com/ioos/compliance-checker/issues/344

File: compliance_checker/cf/cf.py [hunk 1]
```diff
@@ -688,73 +657,73 @@ def check_units(self, ds):
...
-            if v in clim_vars or \
-               v in boundary_vars or \
-               v in metadata_vars or \
-               v in ancillary_vars or \
-               k in container_vars:
+            if name in clim_vars or \
+               name in boundary_vars or \
+               name in metadata_vars or \
+               name in ancillary_vars or \
+               name in container_vars:
```
Comment (daf) [inline compliance_checker/cf/cf.py]: have all of these methods been updated for the name/variable consistency thing?  boundary vars does not look it to me: https://github.com/lukecampbell/compliance-checker/blob/bf45f8bda048c680f0ac4c7ecaa644c529db6d8f/compliance_checker/cf/cf.py#L338-L358

Comment (lukecampbell) [inline compliance_checker/cf/cf.py]: It's in progress. The tests still pass, so what's that say about our tests?

**Comment by lukecampbell** [comment]
I need to update the rest of them like @daf said.

File: compliance_checker/cf/cf.py (hunk 1 above)
Comment (daf) [inline compliance_checker/cf/cf.py]: :laughing:

**Pull request closed without merging**